docker/docker-compose.ci.yml

```diff
@@ -1,14 +1,92 @@
-services:
+version: '3.8'
+
+# Variables
+x-common-vars:
+  - &STACKS_BLOCKCHAIN_COMMIT 49777d3fd73a6dbb610be80c376b7d9389c9871a
+  - &STACKS_API_COMMIT 46e1ae256e474eac98979d0a3df685fe3fa48d6d
+  - &BTC_ADDR miEJtNKa3ASpA19v5ZhvbKTEieYjLpzCYT
+  - &MINER_SEED 9e446f6b0c6a96cf2190e54bcd5a8569c3e386f091605499464389b8d4e0bfc201
+  - &BITCOIN_PEER_PORT 18444
+  - &BITCOIN_RPC_PORT 18443
+  - &BITCOIN_BLOCK_HASH_STREAM_PORT 28332
+  - &BITCOIN_RPC_USER devnet
+  - &BITCOIN_RPC_PASS devnet
+  - &MINE_INTERVAL ${MINE_INTERVAL:-1s}
+  - &MINE_INTERVAL_EPOCH25 ${MINE_INTERVAL_EPOCH25:-5s}
+  - &MINE_INTERVAL_EPOCH3 ${MINE_INTERVAL_EPOCH3:-15s}
+  - &NAKAMOTO_BLOCK_INTERVAL 2
+  - &STACKS_20_HEIGHT ${STACKS_20_HEIGHT:-0}
+  - &STACKS_2_05_HEIGHT ${STACKS_2_05_HEIGHT:-203}
+  - &STACKS_21_HEIGHT ${STACKS_21_HEIGHT:-204}
+  - &STACKS_POX2_HEIGHT ${STACKS_POX2_HEIGHT:-205}
+  - &STACKS_22_HEIGHT ${STACKS_22_HEIGHT:-206}
+  - &STACKS_23_HEIGHT ${STACKS_23_HEIGHT:-207}
+  - &STACKS_24_HEIGHT ${STACKS_24_HEIGHT:-208}
+  - &STACKS_25_HEIGHT ${STACKS_25_HEIGHT:-209}
+  - &STACKS_30_HEIGHT ${STACKS_30_HEIGHT:-232}
+  - &STACKS_31_HEIGHT ${STACKS_31_HEIGHT:-233}
+  - &STACKING_CYCLES ${STACKING_CYCLES:-1}
+  - &POX_PREPARE_LENGTH ${POX_PREPARE_LENGTH:-5}
+  - &POX_REWARD_LENGTH ${POX_REWARD_LENGTH:-20}
+  - &REWARD_RECIPIENT ${REWARD_RECIPIENT:-STQM73RQC4EX0A07KWG1J5ECZJYBZS4SJ4ERC6WN}
+  - &EXIT_FROM_MONITOR 1
+
+# Templates
+x-postgres: &postgres
+  image: postgres:16.6-bookworm@sha256:c965017e1d29eb03e18a11abc25f5e3cd78cb5ac799d495922264b8489d5a3a1
+  stop_grace_period: 5s
+  environment:
+    POSTGRES_USER: postgres
+    POSTGRES_PASSWORD: postgres
+    POSTGRES_DB: signer
+  profiles:
+    - sbtc-signer
+    - sbtc-postgres
 
+x-sbtc-signer: &sbtc-signer
+  build:
+    context: ../
+    dockerfile: docker/sbtc/Dockerfile
+    target: signer
+  entrypoint: "/bin/bash -c '/usr/local/bin/signer -c /signer-config.toml --migrate-db'"
+  environment: &sbtc-signer-environment
+    RUST_LOG: info
+    SIGNER_SIGNER__P2P__LISTEN_ON: tcp://0.0.0.0:4122
+  volumes:
+    - ./sbtc/signer/signer-config.toml:/signer-config.toml
+  profiles:
+    - sbtc-signer
+
+x-stacks-signer: &stacks-signer
+  build:
+    context: stacks
+    args:
+      GIT_COMMIT: *STACKS_BLOCKCHAIN_COMMIT
+  depends_on:
+    - stacks-node
+  volumes:
+    - ./stacks/stacks-signer.toml:/data/config.toml.in
+  environment: &stacks-signer-environment
+    STACKS_NODE_HOST: stacks-node:20443
+    STACKS_SIGNER_ENDPOINT: 0.0.0.0:30000
+  entrypoint:
+    - /bin/bash
+    - -c
+    - |
+      set -e
+      envsubst < config.toml.in > config.toml
+      exec stacks-signer run --config config.toml
+  profiles:
+    - default
+
+# Services
+services:
+  # Existing services from docker-compose.ci.yml
```

Comment:
It seems `bitcoind` and `postgres` here are redundant (we already have a bitcoin node in here, and we could reuse the db of signer-1 already exposed on 5432), why do we need those?